src/os/bluestore/BlueFS.cc

```diff
@@ -1159,11 +1159,20 @@ int BlueFS::_replay(bool noop, bool to_stdout)
       decode(t, p);
     }
     catch (ceph::buffer::error& e) {
-      derr << __func__ << " 0x" << std::hex << pos << std::dec
-           << ": stop: failed to decode: " << e.what()
-           << dendl;
-      delete log_reader;
-      return -EIO;
+      // Multi-block transactions might be incomplete due to unexpected 
+      // power off. Hence let's treat that as a regular stop condition.
+      if (seen_recs && more) {
```

Comment:
seen_recs is set to true a few lines up... so it should probably not be part of this condition?

Reply:
Actually that's seen_recs setting which has to be relocated - one shouldn't set it before decode succeeds. Fixed.